service/javadsl/api/src/main/java/com/lightbend/lagom/javadsl/api/Descriptor.java

```diff
@@ -390,14 +390,14 @@ public String toString() {
     /**
      * Describes a topic call.
      */
-    public static final class TopicCall<Message> {
+    public static final class TopicCall<MESSAGE> {
```

Comment:
>all IDEs highlight generic parameters in some way

Eclipse doesn't highlight generic parameters in any special way. Screenshot attached

>I don't understand the point about confusing with concrete class names

I'm not the only one who thinks this. The Java guidelines echo my point as well that if you use camelcase "it would be difficult to tell the difference between a type variable and an ordinary class or interface name".

> does it matter if it's a generic parameter or not when you're reading the code

I think it's helpful. Knowing that `R` can be literally any type instead of an actual concrete `Request` type makes me think about it differently

>Ctrl+click on it and you'll be taken straight to its definition

True. I was Ctrl+clicking and it wasn't taking me anywhere. It took me a quite awhile to figure out that's because I was looking at the definition. A lot of that was because I just couldn't understand that I was looking at a generic parameter due to the naming. Part of it also was that for awhile I thought Eclipse was broken. Probably other people are less likely to think the latter. I'd been doing so much work setting up Eclipse and encountering all the edge cases and bugs that I thought that this was just another one of them.

![screenshot from 2017-01-31 19 06 48](https://cloud.githubusercontent.com/assets/322311/22494000/924b6ca2-e7e8-11e6-83ce-afd26a7ed95c.png)